### ISSUE TITLE
What is rain_accumulated_final?

### ISSUE THREAD

**Issue by indigodomo** [bpennypacker/WeatherFlow-Indigo-Plugin#14]
Hey Bruce, I don't see a description of that field. What I'm looking for is the daily total precip so that at 11:59PM I can save it off - I'm calculating a rolling 2 day precip window to use to determine if my sprinkler system should run and I'm not sure which field for my Tempest I should use. Thanks for a great plugin!

**Comment by bpennypacker** [comment]
Heh. You've run into the issue that the WeatherFlow documentation is somewhat lacking... If you take a look at the [sky observation (obs_sky) definition](https://weatherflow.github.io/SmartWeather/api/ws.html) in their websocket API it lists fields 14 & 15 as 'Rain Accumulated Final (Rain Check)' and 'Local Daily Rain Accumulation Final (Rain Check)' but doesn't provide any more real information on what they mean. I've gone through a lot of their on-line documentation and never found a definition for precisely what they mean.

I just broke down and asked about these fields in [the WeatherFlow community](https://community.weatherflow.com/t/what-exactly-are-the-rain-accumulated-final-fields/6809) so let's see what they have to say about them!

**Comment by indigodomo** [comment]
Thanks!

**Comment by indigodomo** [comment]
ok, so it sounds like "Local Day Rain Accumulation Final (Rain Check)" is what I want to use at 11:59PM. Now, which state is that stored in?

**Comment by bpennypacker** [comment]
That should be `Daily Rain Accumulation Final` (daily_rain_accumulation_final) in either the Sky or the Tempest devices.

**Comment by indigodomo** [comment]
I'm having a heck of a time reconciling the state values to what's showing in their app. For instance, this is what I'm seeing right now for states (forced the update by bouncing the plugin):

![grab40](https://user-images.githubusercontent.com/5844438/85634906-3fa38800-b642-11ea-8b6c-36cfe78e996e.png)

This is what I'm seeing in the app:

![IMG_5361](https://user-images.githubusercontent.com/5844438/85635065-9ad57a80-b642-11ea-825e-06c2a2b064aa.png)

I can't seem to find matching precip - the app shows .06, the plugin shows 0. The rain_accumulated_final value doesn't correspond to anything I can see - in their app they show 1.81 for yesterday. Lightning strike count is different. (I believe the humidity difference is just because it's been raining off and on so that variance is probably close to correct).

Is there some delay between what's reported via the API and what their app shows? Also, I'm a little concerned that daily_rain_accumulation_final might not be populated at all by 11:59pm, if they aren't updating that regularly. 

My desire for a running precip count is to feed that information into my logic for my sprinklers, so I really need to figure out what the various rain fields mean and when they mean it.

Thoughts?

**Comment by jrickmd** [comment]
For what it’s worth, I’ve never seen the device state for daily_rain_accumulated_final ever change. I watch the Summary data for precip_accum_local_yesterday_final to change and record that as yesterday’s rain in my cumulative data. I don’t use “today’s” data (rain_accumulated_final) for any of my calculations, just stop sprinkling if it becomes non-zero or "precipitation changed to rain” becomes true.

Rick

**Comment by bpennypacker** [comment]
Unfortunately it appears that WeatherFlow uses some non-public information in their app. But on the positive side, another user of this plug-in apparently contacted the WeatherFlow developers and asked a similar question. His email to me just yesterday indicated that he received the following response from the WeatherFlow team:

> Regarding your question about providing a forecast through the API, the answer is "not yet but soon" - everything you see in the app will be available via the API over the next few months.

So the good news is that we should eventually have access to all that data via their public API. Once they've made that available I'll definitely be looking into adding support for it to the plug-in.

**Comment by indigodomo** [comment]
> For what it’s worth, I’ve never seen the device state for daily_rain_accumulated_final ever change. I watch the Summary data for precip_accum_local_yesterday_final to change and record that as yesterday’s rain in my cumulative data. I don’t use “today’s” data (rain_accumulated_final) for any of my calculations, just stop sprinkling if it becomes non-zero or "precipitation changed to rain” becomes true. Rick
> […](#)

Yeah, I just looked at my rolling calculations and they show nothing using that field either. I don't see precip_accum_local_yesterday_final as an available state in my Tempest - Bruce, is that mapped to something else?

**Comment by indigodomo** [comment]
> > Regarding your question about providing a forecast through the API, the answer is "not yet but soon" - everything you see in the app will be available via the API over the next few months.

Hopefully that's not just talking about forecast data...  ;)

**Comment by jrickmd** [comment]
It’s in the summary data… created as a variable automatically but the plugin.’

**Comment by indigodomo** [comment]
> It’s in the summary data… created as a variable automatically but the plugin.’
> […](#)

LOL - didn't even know it did that... 

Unfortunately, that's even more confusing as the non-final version shows 0.030388, but the final version shows 0.0. Yesterday, the app showed .06. I'm beginning to get an uneasy feeling in the pit of my stomach that maybe I shouldn't have invested in this Tempest. Rain reporting was the primary reason (and justification to the spouse) for it's purchase... :/

**Comment by bpennypacker** [comment]
I don't have a Tempest but the original WeatherFlow device that consists of separate Sky & Air devices, and they seem to do a decent job of reporting rain, etc. One thing you might want to try as a test is to find an older Sky/Air station near you and use that Station ID to see how that data compares to what you're seeing from the Tempest. 

If you go to the [WeatherFlow Map](https://tempestwx.com/map/) look for devices near you with a lower numbered Station ID. Go into the plug-in settings and enable debugging, then change the station ID and save the settings. In the Indigo log you'll start to see the raw data as its received from WeatherFlow. You'll see a section titled "devices" that will contain a bunch of entries that look something like this:

            "devices": [
                {
                    "device_id": 53592,
                    "device_meta": {
                        "agl": 0.0,
                        "environment": "indoor",
                        "name": "HB-00015184",
                        "wifi_network_name": ""
                    },
                    "device_type": "HB",
                    "firmware_revision": "119",
                    "hardware_revision": "0",
                    "serial_number": "HB-00015184"
                },
                {
                    "device_id": 53594,
                    "device_meta": {
                        "agl": 10.668,
                        "environment": "outdoor",
                        "name": "Air",
                        "wifi_network_name": ""
                    },
                    "device_type": "AR",
                    "firmware_revision": "23",
                    "hardware_revision": "1",
                    "serial_number": "AR-00019739"
                },

Each of those blocks represents one device for the particular station. The hub that connects to the internet will be identified by the device_type "HB". A Sky device is "SK", and an Air device is "AR". Those are the devices prior to the release of the Tempest. The Tempest will identify itself as "ST". Once you've found an older Sky/Air station try adding the Sky device to Indigo, as that's the device that detects rain. If you leave debugging enabled you should see the raw data for the Sky device in the log as it's received.

If/when I have some time I'll try running some tests with different stations and see if the results are comparable between the older Sky/Air devices and the newer Tempest. (I regularly pick random devices around the country to test with depending on if I need data that includes rain, lightning, etc. and where the radar at weather.com shows activity. It's also how I tested the Tempest device without having to go buy one myself.)
